ecmascript/parser/src/parser/class_and_fn.rs

```diff
@@ -736,6 +734,9 @@ impl<'a, I: Tokens> Parser<I> {
         if is_constructor(&key) {
             syntax_error!(key.span(), SyntaxError::PropertyNamedConstructor);
         }
+        if declare && key.is_left() {
+            syntax_error!(key.span(), SyntaxError::DeclarePrivateIdentifier);
```

Comment:
What's the recommended way to add a test for this?

Reply:
I think creating a file at ecmascript/parser/tests/typescript/issue-894/input.ts would be the best way

Reply:
I ended up putting it in the typescript-errors folder because I couldn't get it working in the typescript folder after trying a bit.

Reply:
I'm sorry. I thought you was talking about correct case.
(I'm sleepy as it's over 02 am.)